superbench/common/utils/nvidia_helper.py

```diff
@@ -5,18 +5,180 @@
+    except nvml.NVMLError as err:
+        logger.error('Get device memory failed: {}'.format(str(err)))
+        return None, None
+    return mem.used, mem.total
+
+
+def get_device_row_remapped_info(idx):
+    """Get the row remapped information of device.
+
+    The command 'nvidia-smi -i idx -q' contains the following output:
+        Remapped Rows
+            Correctable Error                 : 0
+            Uncorrectable Error               : 0
+            Pending                           : No
+            Remapping Failure Occurred        : No
+            Bank Remap Availability Histogram
+                Max                           : 640 bank(s)
+                High                          : 0 bank(s)
+                Partial                       : 0 bank(s)
+                Low                           : 0 bank(s)
+                None                          : 0 bank(s)
+        Temperature
+            GPU Current Temp                  : 36 C
+
+    Args:
+        idx (int): device index.
+
+    Return:
+        remapped_metrics (dict): the row remapped information, None means failed to get the data.
+    """
+    output = run_command('nvidia-smi -i {} -q'.format(idx))
+    if output.returncode == 0:
+        begin = output.stdout.find('Remapped Rows')
+        end = output.stdout.find('Temperature', begin)
+        if begin != -1 and end != -1:
+            remapped_info = output.stdout[begin:end]
+            remapped_info = remapped_info.split('\n')
+            remapped_info = [item for item in remapped_info if ':' in item]
+            remapped_metrics = dict()
+            for item in remapped_info:
+                key_value = item.split(':')
+                key = 'gpu_remap_' + key_value[0].lower().strip().replace(' ', '_')
+                value = key_value[1].replace('bank(s)', '').strip()
+                try:
+                    value = int(value)
+                    remapped_metrics[key] = value
+                except BaseException:
+                    continue
+
+            return remapped_metrics
+
+    return None
+
+
+def get_device_ecc_error(idx):
+    """Get the ecc error information of device.
+
+    The command 'nvidia-smi dmon -i 0 -s e -c 1' gets the following output:
+        # gpu sbecc dbecc   pci
+        # Idx  errs  errs  errs
+            0     0     0     0
+
+    Args:
+        idx (int): device index.
+
+    Return:
+        sbecc (int): the count of single bit ecc error, None means failed to get the data.
+        dbecc (int): the count of double bit ecc error, None means failed to get the data.
+    """
+    output = run_command('nvidia-smi dmon -i {} -s e -c 1'.format(idx))
+    if output.returncode == 0:
+        content = output.stdout.splitlines()
+        if len(content) == 3:
+            ecc = list(filter(None, content[2].split(' ')))
+            if len(ecc) == 4:
+                sbecc = int(ecc[1])
+                dbecc = int(ecc[2])
```

Comment:
what if they are 'N/A' on geforce cards?

Reply:
use `nvmlDeviceGetDetailedEccErrors`?

Reply:
`nvmlDeviceGetDetailedEccErrors` is labelled as deprecated:
https://github.com/fbcotter/py3nvml/blob/0aa41c0940c7d971381fddb8382266a819be9f75/py3nvml/py3nvml.py#L3437

> 
>      * @deprecated   This API supports only a fixed set of ECC error locations
>      *               On different GPU architectures different locations are supported
>      *               See \ref nvmlDeviceGetMemoryErrorCounter

I will use `nvmlDeviceGetMemoryErrorCounter` instead, which use correctable for single bit and uncorrectable for double bit.